compiler/rustc_lint/src/late.rs

```diff
@@ -214,15 +213,12 @@ impl<'tcx, T: LateLintPass<'tcx>> hir_visit::Visitor<'tcx> for LateContextAndPas
         })
     }
 
-    fn visit_ty(&mut self, t: &'tcx hir::Ty<'tcx>) {
-        lint_callback!(self, check_ty, t);
+    fn visit_ty(&mut self, t: &'tcx hir::Ty<'tcx, AmbigArg>) {
+        // FIXME(thispr): we dont call `check_ty` on infer vars
```

Comment:
We no longer call `check_ty` for late lint passes on infer args I should probably make `check_ty` accept an ambig ty so its clear you cant access infer vars on it and then if anything does need infer vars, add a `visit_infer`